docs/built-in-functions.rst

```diff
@@ -0,0 +1,237 @@
+.. index:: function, built-in;
+
+.. _built_in_functions:
+
+***********************
+Built in Functions
+***********************
+
+Viper contains a set amount of built in functions that would be timely and/or unachievable to write in Viper.
+
+.. _functions:
+
+Functions
+=========
+The first list dictates each argument and the second sublist lists the types allowed as input.
+
+* **floor**
+::
+
+  def floor(a) -> b:
+    """
+    :param a: value to round down
+    :type a: either decimal or num
+
+    :output b: interger
+    """
+Rounds a decimal down to the nearest integer.
+
+* **decimal**
+::
+
+  def decimal(a) -> b:
+    """
+    :param a: value to turn into decimal
+    :type a: either decimal or num
+
+    :output b: decimal
+    """
+Turns a number into a decimal.
+
+* **as_unitless_number**
+::
+
+  def as_unitless_number(a) -> b:
+    """
+    :param a: value to remove units from
+    :type a: either decimal or num
+
+    :output b: either decimal or num
+    """
+Turns a ``num`` or ``decimal`` with units into one without units (used for assignment and math).
+
+* **as_num128**
+::
+
+  def as_num128(a) -> b:
+    """
+    :param a: value to turn into int128
+    :type a: either num, bytes32, num256, or bytes
+
+    :output b: num
+    """
+Turns input into a ``num`` (int128).
+
+* **as_num256**
+::
+
+  def as_num256(a) -> b:
+    """
+    :param a: value to turn into num256
+    :type a: either num_literal, num, bytes32, or address
+
+    :output b: num256
+    """
+Turns input into a ``num256`` (uint256).
+
+* **as_bytes32**
+::
+
+  def as_bytes32(a) -> b:
+    """
+    :param a: value to turn into bytes32
+    :type a: either num, num256, address
+
+    :output b: bytes32
+    """
+Turns input into a ``bytes32``.
+
+* **slice**
+::
+
+  def slice(a, start=b, length=c) -> d:
+    """
+    :param a: bytes to be sliced
+    :type a: either bytes or bytes32
+    :param b: start position of the slice
+    :type b: num
+    :param c: length of the slice
+    :type c: num
+
+    :output d: bytes
+    """
+Takes a a list of bytes, the start bytes to take and the length to take and returns the associated chunk.
+
+* **len**
+::
+
+  def len(a) -> b:
+    """
+    :param a: value to get the length of
+    :type a: bytes
+
+    :output b: num
+    """
+Returns the length of a given list of bytes.
+
+* **concat**
```

Comment:
concat appears to lack the signature annotation (Is this problematic?)

Reply:
I don't think it's a problem, it doesn't have signature annotations  because it takes a dynamic number of arguments.

Reply:
Thought that might be the case. On inspection of the code it looks ok, as concat checks its own argument semantics.